machine/v1beta1/types_azureprovider.go

```diff
@@ -229,7 +279,8 @@ type DiskSettings struct {
 
 // ManagedDiskParameters is the parameters of a managed disk.
 type ManagedDiskParameters struct {
-	// StorageAccountType is the storage account type to use. Possible values include "Standard_LRS" and "Premium_LRS".
+	// StorageAccountType is the storage account type to use.
+	// Possible values include "Standard_LRS", "Premium_LRS" and, only for Data Disks, "UltraSSD_LRS."
```

Comment:
Instead of having conditional validation, create a new type.  Starting with drift suggests that drift will continue.

Alternatively, let OSDisks use UltraSSD_LRS

Reply:
Unfortunately we can't use Ultra Disks (`UltraSSD_LRS`) on OS Disks.
So would you split them into `OSDiskManagedDiskParameters` and `DataDiskManagedDiskParameters` for example?

Reply:
> So would you split them into OSDiskManagedDiskParameters and DataDiskManagedDiskParameters for example?

Yes.  They have different values, validation, and documentation.

Reply:
Ok I kept the original `ManagedDiskParameters` to be used only for OS Disks, and created a new `DataDiskManagedDiskParameters`.

Reply:
Now changed to `OSDiskManagedDiskParameters`.